src/fsharp/utils/FileSystem.fs

```diff
@@ -604,6 +595,28 @@ module public StreamExtensions =
             let encoding = defaultArg encoding Encoding.UTF8
             s.ReadLines(encoding) |> Seq.toArray
 
+        member s.AsByteMemory() : ByteMemory =
+            match s with
+            | :? MemoryMappedViewStream as mmvs ->
+                let safeHolder =
+                    { new obj() with
+                        override x.Finalize() =
+                            (x :?> IDisposable).Dispose()
+                      interface IDisposable with
+                        member x.Dispose() =
+                            GC.SuppressFinalize x
+                            mmvs.Dispose() }
+                let length = mmvs.Length
+                RawByteMemory(
+                    NativePtr.ofNativeInt (mmvs.SafeMemoryMappedViewHandle.DangerousGetHandle()),
+                    int length,
+                    safeHolder) :> ByteMemory
+
+            | _ ->
+                let bytes = s.ReadAllBytes()
```

Comment:
This is interesting that if the stream is not memory-mapped, then we will copy the contents. Could we put a comment describing this behavior and why?

Reply:
It pretty much repeats the old behaviour - if we use mmaped files, we don't copy and just expose it via the handle, but in case if we don't (i.e. reading smaller files OR if we run on mono) - we were just reading it via File.ReadAllBytes().

So, here it's pretty much the same, if we are working with the streamview from mmf, we wrap it in RawByteMemory (which does zero copy). But when we use any other stream (FileStream, MemoryStream, etc) - we just read everything from it and expose via ByteArrayMemory.

I will put the comment explaining this.